OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/AnalyticsUploadHelper.java

```diff
@@ -44,44 +45,30 @@ enum Result {
                 events = eventsResult.first;
                 maxEventId = eventsResult.second;
             }
-        }
-        catch (SQLiteException e) {
+        } catch (SQLiteException e) {
             e.printStackTrace();
             return Result.FAILED_RETRY_LATER;
         }
+        catch (Optimobile.PartialInitialisationException e) {
+            return Result.FAILED_NO_RETRY;
+        }
 
         return Result.SUCCESS;
     }
 
-    private boolean flushBatchToNetwork(Context context, ArrayList<JSONObject> events, long maxEventId) {
+    private boolean flushBatchToNetwork(Context context, ArrayList<JSONObject> events, long maxEventId) throws Optimobile.PartialInitialisationException {
```

Comment:
Does it throw? seems like it catches this type here and doesnt rethrow (the try autocloseble ends up in the catch block

Reply:
catch in `flushBatchToNetwork` is only for IOException, this one is caught in `flushEvents`